kornia/filters/filter.py

```diff
@@ -71,16 +91,119 @@ def filter2D(input: torch.Tensor, kernel: torch.Tensor,
+        border_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'replicate'``.
+        normalized (bool): If True, kernel will be L1 normalized.
+
+    Return:
+        torch.Tensor: the convolved tensor of same size and numbers of channels
+        as the input with shape :math:`(B, C, D, H, W)`.
+
+    Example:
+        >>> input = torch.tensor([[[
+        ...    [[0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.]],
+        ...
+        ...    [[0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 5., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.]],
+        ...
+        ...    [[0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.]]]])
+        >>> kernel = torch.ones(1, 3, 3, 3)
+        >>> kornia.filter3D(input, kernel)
+        torch.tensor([[[[[0., 0., 0., 0., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 0., 0., 0., 0.]],
+                        [[0., 0., 0., 0., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 0., 0., 0., 0.]],
+                        [[0., 0., 0., 0., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 0., 0., 0., 0.]]]])
+    """
+    testing.check_is_tensor(input)
+    testing.check_is_tensor(kernel)
+
+    if not isinstance(border_type, str):
+        raise TypeError("Input border_type is not string. Got {}"
+                        .format(type(kernel)))
+
+    if not len(input.shape) == 5:
+        raise ValueError("Invalid input shape, we expect BxCxDxHxW. Got: {}"
+                         .format(input.shape))
+
+    if not len(kernel.shape) == 4:
+        raise ValueError("Invalid kernel shape, we expect BxDxHxW. Got: {}"
+                         .format(kernel.shape))
+
+    borders_list: List[str] = ['constant', 'replicate', 'circular']
+    if border_type not in borders_list:
+        raise ValueError("Invalid border_type, we expect the following: {0}."
+                         "Got: {1}".format(borders_list, border_type))
+
+    # prepare kernel
+    b, c, d, h, w = input.shape
+    tmp_kernel: torch.Tensor = kernel[None].type_as(input)
+
+    if normalized:
+        bk, dk, hk, wk = kernel.shape
+        tmp_kernel = normalize_kernel2d(tmp_kernel.view(
```

Comment:
Shall we rename/reimplement the ```normalize_kernel2d``` into just ```normalize_kernel```? I think this conversion step shall happen inside ```normalize_kernel``` than here.

Reply:
lets leave it for a future PR ?